application/utils.inc.php

```diff
@@ -2316,4 +2316,24 @@ public static function ToCamelCase($sInput)
 	{
 		return str_replace(' ', '', ucwords(strtr($sInput, '_-', '  ')));
 	}
+
+	/**
+	 * @param \cmdbAbstractObject $oCmdbAbstract
+	 * @param \Exception $oException
+	 *
+	 * @throws \Exception
+	 */
+	public static function EnrichRaisedException($oCmdbAbstract, $oException)
+	{
+		if (is_null($oCmdbAbstract) ||
+			! is_a($oCmdbAbstract, \cmdbAbstractObject::class))
+		{
+			throw $oException;
+		}
+
+		$sMessage = $oException->getMessage() . " (" . get_class($oCmdbAbstract) . "::" . $oCmdbAbstract->Get('id') . ")";
```

Comment:
Why not use predefined `__toString` method?

> HTML String representation of the object
> 
> Only a few meaningful information will be returned.
> This representation is for debugging purposes, and is subject to change.
> The returned string is raw HTML

```suggestion
		$sMessage = $oCmdbAbstract . " " . $oException->getMessage();
```

Reply:
good idea for code refactoring. but __toString() currently returns html content. isn't it weard in logs?

Reply:
True, but that is a weirdness of the `__toString` method.

Note that most of these messages do get displayed on the UI (Hence the existence of `CoreException::getTraceAsHtml()`)

Reply:
i will used __tostring() as recommanded.

Reply:
I didn't recommend it. I was just asking if this wouldn't be the better way.